### PR TITLE
add compat tests for reconnects and progress

### PR DESCRIPTION
**Pull request by anurse**
More stuff for #206 

Adds compat tests for progress reporting, and reconnects (but our "force reconnect" technique only really works for WebSockets. I verified that SignalR 2.x has the same problem)

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs [hunk 1]
```diff
@@ -127,6 +129,42 @@ public ChatHubTests(ServerFixture fixture)
...
+
+                Assert.Equal(new[] { 0, 1, 2, 3, 4 }, client1.ReceivedProgress.ToArray());
+            }
+        }
+
+
+        [Fact]
+        public async Task Can_recover_from_reconnect()
+        {
+            // Our technique for forcing a reconnect only works for WebSockets right now...
```
Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs]: I wonder how we tested reconnects for other transports in the old server then...

Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs]: Yeah, I'm going to take a look at that later. Still want to test reconnects for those other transports.

Comment (moozzyk) [inline test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs]: In that case - :ship::it:

Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.CompatTests/ChatHubTests.cs]: I found a way to test reconnects for the HTTP-based transports, will roll that in to the next PR (which adds tests for raw connections).

**Pull request merged**